source/simulator/core.cc

```diff
@@ -1687,12 +1687,28 @@ namespace aspect
       signals.pre_refinement_store_user_data(triangulation);
 
 
-      {
 
-      }
       exchange_refinement_flags();
 
       triangulation.prepare_coarsening_and_refinement();
+      bool any_flags_set = false;
+      {
+        for (const auto &cell:dof_handler.active_cell_iterators())
+          {
+            if (cell->refine_flag_set() || cell->coarsen_flag_set())
+              {
+                any_flags_set = true;
+                break;
+              }
+          }
+      }
+      const bool mesh_changed = Utilities::MPI::max(any_flags_set?1:0,mpi_communicator) == 1 ? true : false;
+      if (!mesh_changed)
+        {
+          pcout << "\tSkipping mesh refinement, because the mesh did not change." << std::endl;
```

Comment:
This needs less indentation (the regular output when cells are refined has no indentation at all), and another new line at the end (to be consistent with what we do in case of mesh refinement).

Reply:
I guess you are voting for 0 indentation? Ok.